lib/netext/httpext/request.go

```diff
@@ -187,17 +187,17 @@ func MakeRequest(ctx context.Context, state *lib.State, preq *ParsedHTTPRequest)
 		}
 	}
 
-	tags := state.CloneTags()
+	tags := state.Tags.BranchOut()
 	// Override any global tags with request-specific ones.
 	for k, v := range preq.Tags {
-		tags[k] = v
+		tags.AddTag(k, v)
```

Comment:
nit: same nit as in the case in k6/metrics - this potentially can be really bad with a lot of tags and them being walked randomly.